### PR TITLE
Add exception handling mode

### PR DESCRIPTION
**Pull request by ashulenko**
- Changing naming of ClickHouse (converting letter "h" to uppercase);
- Add exception handling with CompletableFuture;
- ClickHouse sending exception handling config was added:
1. if `ignoring-clickhouse-sending-exception-enabled` is true, exception while clickhouse sending is ignored and failed data automatically goes to the disk.
2. if `ignoring-clickhouse-sending-exception-enabled` is false, clickhouse sending exception thrown in "main" thread (thread which called ClickhHouseSink::invoke) and data also goes to the disk.

### CODE REVIEW

File: src/test/java/ru/ivi/opensource/flinkclickhousesink/ClickHouseWriterTest.java [hunk 1]
```diff
@@ -174,23 +180,119 @@ public void highConcurrentTest() throws Exception {
...
+                Sink<String> sink = buffers.get(id);
                 String csv;
                 if (id % 2 != 0) {
                     csv = "(10, 'title', 'container', 'drm', 'quality')";
                 } else {
                     csv = "(11, 'title', 111)";
                 }
-                sinkBuffer.put(csv);
+                sink.put(csv);
+            } catch (InterruptedException e) {
```
Comment (mchernyakov) [inline src/test/java/ru/ivi/opensource/flinkclickhousesink/ClickHouseWriterTest.java:191]: could be:
```java
catch (InterruptedException | ExecutionException e)
```

Comment (ashulenko) [reply src/test/java/ru/ivi/opensource/flinkclickhousesink/ClickHouseWriterTest.java]: done

File: src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ClickHouseWriter.java [hunk 2]
```diff
@@ -88,47 +96,77 @@ public void put(ClickhouseRequestBlank params) {
         }
     }
 
+    private void waitUntilAllFuturesDone() {
+        logger.info("Wait until all futures are done or completed exceptionally. Futures size: {}", futures.size());
+        try {
+            while (unprocessedRequestsCounter.get() > 0 || !futures.isEmpty()) {
+                logger.debug("Futures size: {}.", futures.size());
```
[the remark this answers is not in the document]
Comment (ashulenko) [reply src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ClickHouseWriter.java]: done

File: src/main/java/ru/ivi/opensource/flinkclickhousesink/util/FutureUtil.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package ru.ivi.opensource.flinkclickhousesink.util;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public final class FutureUtil {
```
[the remark this answers is not in the document]
Comment (ashulenko) [reply src/main/java/ru/ivi/opensource/flinkclickhousesink/util/FutureUtil.java:6]: done

File: src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ExceptionsThrowableSink.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+public class ExceptionsThrowableSink<T> implements Sink<T> {
+    private final ClickHouseSinkBuffer clickHouseSinkBuffer;
+
+    public ExceptionsThrowableSink(ClickHouseSinkBuffer buffer) {
+        this.clickHouseSinkBuffer = buffer;
+    }
+
+    @Override
+    public void put(T message) throws ExecutionException, InterruptedException {
+        clickHouseSinkBuffer.put((String) message);
```
[the remark this answers is not in the document]
Comment (ashulenko) [reply src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/ExceptionsThrowableSink.java]: This is good question. No, it doesn't. 

Remove generic interface Sink and generic class ExceptionsThrowableSink.

File: src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/UnexceptionableSink.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+public class UnexceptionableSink<T> implements Sink<T> {
+    private final ClickHouseSinkBuffer clickHouseSinkBuffer;
+
+    public UnexceptionableSink(ClickHouseSinkBuffer buffer) {
+        this.clickHouseSinkBuffer = buffer;
+    }
+
+    @Override
+    public void put(T message) {
+        clickHouseSinkBuffer.put((String) message);
```
[the remark this answers is not in the document]
Comment (ashulenko) [reply src/main/java/ru/ivi/opensource/flinkclickhousesink/applied/UnexceptionableSink.java]: This is good question. No, it doesn't. 

Remove generic interface Sink and generic class UnexceptionableSink

File: README.md [hunk 6]
```diff
@@ -48,17 +48,21 @@ common and for each sink in you operators chain.
  
  `clickhouse.sink.retries` - max number of retries,
  
- `clickhouse.sink.failed-records-path`- path for failed records.
+ `clickhouse.sink.failed-records-path`- path for failed records,
+ 
+ `clickhouse.sink.ignoring-clickhouse-sending-exception-enabled` - boolean parameter responsible for raising (false) or not (true) ClickHouse sending exception in main thread. 
+ if `ignoring-clickhouse-sending-exception-enabled` is true, exception while clickhouse sending is ignored and failed data automatically goes to the disk.
```
[the remark this answers is not in the document]
Comment (ashulenko) [reply README.md:54]: This parameter is required [(check).](https://github.com/ivi-ru/flink-clickhouse-sink/blob/add_exception_handling_mode/src/main/java/ru/ivi/opensource/flinkclickhousesink/model/ClickHouseSinkCommonParams.java#L22) That's why there is no default value. 
Documentation changed.

**Review by mchernyakov — APPROVED** [review]

**Pull request merged**